drivers/memc/memc_stm32_xspi_psram.c

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_xspi_psram
+
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(memc_stm32_xspi_psram, CONFIG_MEMC_LOG_LEVEL);
+
+#define STM32_XSPI_NODE DT_INST_PARENT(0)
+
+/* Memory registers definition */
+#define MR0		0x00000000U
+#define MR1		0x00000001U
+#define MR2		0x00000002U
+#define MR3		0x00000003U
+#define MR4		0x00000004U
+#define MR8		0x00000008U
+
+/* Memory commands */
+#define SYNC_READ_CMD	0x00
+#define SYNC_WRITE_CMD	0x80
+#define BURST_READ_CMD	0x20
+#define BURST_WRITE_CMD	0xA0
+#define READ_REG_CMD	0x40
+#define WRITE_REG_CMD	0xC0
+#define RESET_CMD	0xFF
+
+/* Memory default dummy clocks cycles */
+#define DUMMY_CLK_CYCLES_READ	6
+#define DUMMY_CLK_CYCLES_WRITE	6
+
+
+#define BUFFERSIZE	10240
+
+struct memc_stm32_xspi_psram_config {
+	const struct pinctrl_dev_config *pcfg;
+	const struct stm32_pclken pclken;
+	const struct stm32_pclken pclken_ker;
+	const struct stm32_pclken pclken_mgr;
+	size_t memory_size;
+};
+
+struct memc_stm32_xspi_psram_data {
+	XSPI_HandleTypeDef hxspi;
+};
+
+uint32_t ap_memory_write_reg(XSPI_HandleTypeDef *hxspi, uint32_t address, uint8_t *value)
+{
+	XSPI_RegularCmdTypeDef cmd = {0};
+
+	/* Initialize the write register command */
+	cmd.OperationType = HAL_XSPI_OPTYPE_COMMON_CFG;
+	cmd.Instruction = WRITE_REG_CMD;
+	cmd.InstructionMode = HAL_XSPI_INSTRUCTION_8_LINES;
+	cmd.InstructionWidth = HAL_XSPI_INSTRUCTION_8_BITS;
+	cmd.InstructionDTRMode = HAL_XSPI_INSTRUCTION_DTR_DISABLE;
+	cmd.Address = address;
+	cmd.AddressMode = HAL_XSPI_ADDRESS_8_LINES;
+	cmd.AddressWidth = HAL_XSPI_ADDRESS_32_BITS;
+	cmd.AddressDTRMode = HAL_XSPI_ADDRESS_DTR_ENABLE;
+	cmd.AlternateBytesMode = HAL_XSPI_ALT_BYTES_NONE;
+	cmd.DataMode = HAL_XSPI_DATA_8_LINES;
+	cmd.DataLength = 2;
```

Comment:
`2U`?  (ditto at line 110 and few other places below)

Reply:
Done